x/tokenomics/keeper/settle_session_accounting_test.go

```diff
@@ -0,0 +1,273 @@
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+	"github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+// TODO_IN_THIS_PR: Evaluate what other tests we need to add in this iteration.
+
+func TestSettleSessionAccounting_ValidAccounting(t *testing.T) {
+	t.Skip("TODO_BLOCKER(@Olshansk): Add E2E and integration tests so we validate the actual state changes of the bank & account keepers.")
+	// Assert that `suppliertypes.ModuleName` account module balance is *unchanged*
+	// Assert that `supplierAddress` account balance has *increased* by the appropriate amount
+	// Assert that `supplierAddress` staked balance is *unchanged*
+	// Assert that `apptypes.ModuleName` account module balance is *unchanged*
+	// Assert that `applicationAddress` account balance is *unchanged*
+	// Assert that `applicationAddress` staked balance has decreased by the appropriate amount
+}
+
+func TestSettleSessionAccounting_AppStakeTooLow(t *testing.T) {
+	t.Skip("TODO_BLOCKER(@Olshansk): Add E2E and integration tests so we validate the actual state changes of the bank & account keepers.")
+	// Assert that `suppliertypes.Address` account balance has *increased* by the appropriate amount
+	// Assert that `applicationAddress` account staked balance has gone to zero
+	// Assert on whatever logic we have for slashing the application or other
+}
+
+func TestSettleSessionAccounting_AppNotFound(t *testing.T) {
+	keeper, ctx, _, supplierAddr := testkeeper.TokenomicsKeeper(t)
+	wctx := sdk.WrapSDKContext(ctx)
+
+	// The base claim whose root will be customized for testing purposes
+	claim := suppliertypes.Claim{
+		SupplierAddress: supplierAddr,
+		SessionHeader: &sessiontypes.SessionHeader{
+			ApplicationAddress: sample.AccAddress(), // Random address
+			Service: &sharedtypes.Service{
+				Id:   "svc1",
+				Name: "svcName1",
+			},
+			SessionStartBlockHeight: 1,
+			SessionId:               "1",
+			SessionEndBlockHeight:   5,
+		},
+		RootHash: smstRootWithSum(42),
+	}
+
+	err := keeper.SettleSessionAccounting(wctx, &claim)
+	require.Error(t, err)
+	require.ErrorIs(t, err, types.ErrTokenomicsApplicationNotFound)
+}
+
+func TestSettleSessionAccounting_InvalidRoot(t *testing.T) {
+	keeper, ctx, appAddr, supplierAddr := testkeeper.TokenomicsKeeper(t)
+	wctx := sdk.WrapSDKContext(ctx)
+
+	// Define test cases
+	testCases := []struct {
+		desc        string
+		root        []byte // smst.MerkleRoot
+		errExpected bool
+	}{
+		{
+			desc:        "Nil Root",
+			root:        nil,
+			errExpected: true,
+		},
+		{
+			desc:        "Less than 40 bytes",
+			root:        []byte("less than 40 bytes"),
+			errExpected: true,
+		},
+		// TODO_IN_THIS_PR_DISCUSS: Should this error? Should it be on the SMT side?
+		// {
+		// 	desc:      "More than 40 bytes",
+		// 	root:      []byte("more than 40 bytes, meaning this is way too long"),
+		// 	errExpected: true,
+		// },
+		// TODO_IN_THIS_PR_DISCUSS: Should this error? Should it be on the SMT side?
```

Comment:
I'm not sure what this means, `make([]byte, 40)` aka a 40-byte slice with all 0s represents an empty tree with a 0 sum. This is valid iff. the node did no work in the session.